apps/spruce/src/pages/version/Tabs/VersionTiming/index.tsx

```diff
@@ -0,0 +1,148 @@
+import { useEffect } from "react";
+import { useQuery } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Body, BodyProps, H2 } from "@leafygreen-ui/typography";
+import { useLocation } from "react-router-dom";
+import { fontSize, size } from "@evg-ui/lib/constants/tokens";
+import { useToastContext } from "@evg-ui/lib/context/toast";
+import { useVersionAnalytics } from "analytics";
+import TableControl from "components/Table/TableControl";
+import { DEFAULT_POLL_INTERVAL } from "constants/index";
+import { PaginationQueryParams, TableQueryParams } from "constants/queryParams";
+import {
+  SortDirection,
+  TaskSortCategory,
+  VersionTaskDurationsQuery,
+  VersionTaskDurationsQueryVariables,
+} from "gql/generated/types";
+import { VERSION_TASK_DURATIONS } from "gql/queries";
+import { usePolling } from "hooks";
+import { useQueryParams } from "hooks/useQueryParam";
+import { PatchTasksQueryParams } from "types/task";
+import { useQueryVariables } from "../useQueryVariables";
+import {
+  transformTaskDurationDataToVariantGanttChartData,
+  transformTaskDurationDataToTaskGanttChartData,
+} from "./utils";
+import VersionTimingGraph from "./VersionTimingGraph";
+
+interface Props {
+  versionId: string;
+  taskCount: number;
+}
+
+const VersionTiming: React.FC<Props> = ({ taskCount, versionId }) => {
+  const dispatchToast = useToastContext();
+  const { search } = useLocation();
+
+  const defaultSort = `${TaskSortCategory.Duration}:${SortDirection.Desc}`;
```

Comment:
Nit. you can move this outside the component like we do in TaskDuration since it never changes.